core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/internal/DependencyDocument.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2023 B2i Healthcare Pte Ltd, http://b2i.sg
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.b2international.snowowl.core.internal;
+
+import java.util.Comparator;
+import java.util.Objects;
+
+import com.b2international.snowowl.core.ResourceURIWithQuery;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * @since 8.12.0
+ */
+public final class DependencyDocument implements Comparable<DependencyDocument> {
```

Comment:
This class might need extra annotations or dummy setters for the derived fields if the "ignore unknown properties" option is not enabled.

Reply:
Tests are running just fine, so not sure about this.